### PR TITLE
support creating an image from a volume

### PR DESCRIPTION
**Pull request by bverschueren**
##### SUMMARY
Support creating an image from a volume

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
os_image

### CODE REVIEW

File: lib/ansible/modules/cloud/openstack/os_image.py [hunk 1]
```diff
@@ -132,8 +164,13 @@ def main():
         ramdisk=dict(default=None),
         kernel=dict(default=None),
         properties=dict(type='dict', default={}),
+        volume=dict(default=None),
         state=dict(default='present', choices=['absent', 'present']),
     )
+
+    required_one_of = [['filename', 'volume']],
```
Comment (jamescassell) [inline lib/ansible/modules/cloud/openstack/os_image.py]: Filename wasn't required before. Why should be required now?

**Review by jamescassell — APPROVED** [review]
Just one comment.

File: lib/ansible/modules/cloud/openstack/os_image.py (hunk 1 above)
Comment (bverschueren) [reply lib/ansible/modules/cloud/openstack/os_image.py]: You're right, should be just mutually exclusive.

File: lib/ansible/modules/cloud/openstack/os_image.py [hunk 2]
```diff
@@ -132,8 +164,12 @@ def main():
         ramdisk=dict(default=None),
         kernel=dict(default=None),
         properties=dict(type='dict', default={}),
+        volume=dict(default=None),
         state=dict(default='present', choices=['absent', 'present']),
     )
+
+    mutually_exclusive = [['filename', 'volume']],
```
Comment (cloudnull) [inline lib/ansible/modules/cloud/openstack/os_image.py]: Is this supposed to be used somewhere?

``` python
In [1]: mutually_exclusive = [['filename', 'volume']],

In [2]: mutually_exclusive
Out[2]: ([['filename', 'volume']],)
```

This object is just a tuple with a nested list of lists. I suspect we need to load this into some function to make it actually mutually exclusive?

**Review by cloudnull — CHANGES_REQUESTED** [review]
see above

File: lib/ansible/modules/cloud/openstack/os_image.py (hunk 2 above)
Comment (bverschueren) [reply lib/ansible/modules/cloud/openstack/os_image.py]: Indeed, it needs to be passed to the AnsibleModule initialization to have any effect.
It's fixed now.

**Comment by bverschueren** [comment]
ready_for_review

**Review by cloudnull — APPROVED** [review]
shipit

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@bverschueren thanks for implementing this!
@cloudnull @jamescassell thanks for reviewing!

**Pull request merged**